src/server/cluster/cluster_family_test.cc

```diff
@@ -739,27 +648,24 @@ TEST_F(ClusterFamilyTest, FlushSlots) {
                                                   _, "total_writes", _, "memory_bytes", _)))));
 }
 
+TEST_F(ClusterFamilyTest, FlushSlotsAndImmediatelySetValue) {
+  for (string_view count : {"1", "10", "100", "1000", "10000", "100000"}) {
+    ConfigSingleNodeCluster(GetMyId());
+
+    EXPECT_EQ(Run({"debug", "populate", count, "key", "4"}), "OK");
+    EXPECT_THAT(Run({"cluster", "keyslot", "key:0"}), IntArg(2592));
+    EXPECT_EQ(Run({"dflycluster", "flushslots", "2592", "2592"}), "OK");
+    // key:0 should have been removed, so APPEND will end up with key:0 == ZZZZ
+    EXPECT_THAT(Run({"append", "key:0", "ZZZZ"}), IntArg(4));
+    EXPECT_EQ(Run({"get", "key:0"}), "ZZZZ");
+
```

Comment:
maybe you can also run dbsize and dflycluster getslotinfo slots 2592 2592 before after flushslots and check that actually deleted all slot 2592 and only them